flowzone.yml

```diff
@@ -2365,114 +2317,93 @@ jobs:
+            npm i
           fi
 
       - name: Run build
+        if: steps.check_engine.outcome == 'success'
         run: npm run build --if-present
 
       - name: Run tests
-        if: inputs.pseudo_terminal != true
+        if: inputs.pseudo_terminal != true && steps.check_engine.outcome == 'success'
         run: npm test
 
       - name: Run tests (pseudo-tty)
-        if: inputs.pseudo_terminal == true
+        if: inputs.pseudo_terminal == true && steps.check_engine.outcome == 'success'
         shell: script -q -e -c "bash --noprofile --norc -eo pipefail -x {0}" /tmp/test-session
         run: npm test
 
       - name: Run pack
-        if: needs.is_npm.outputs.npm_private != 'true' && steps.node_versions.outputs.max == matrix.node_version
+        if: needs.is_npm.outputs.npm_private != 'true' && steps.check_engine.outcome == 'success'
         run: |
           mkdir ${{ runner.temp }}/npm-pack && npm pack --pack-destination=${{ runner.temp }}/npm-pack
 
           # FIXME: workaround when `npm pack` for npm 6.x dumps tarball into the current directory because it has no `--pack-destination` flag
           [[ "$(npm --version)" =~ ^6\..* ]] && find . -maxdepth 1 -name '*.tgz' -exec mv {} ${{ runner.temp }}/npm-pack \; || true
 
       - name: Upload artifact
-        if: needs.is_npm.outputs.npm_private != 'true' && steps.node_versions.outputs.max == matrix.node_version
+        if: needs.is_npm.outputs.npm_private != 'true' && steps.check_engine.outcome == 'success'
         uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # v4.3.4
         with:
-          name: npm-${{ github.event.pull_request.head.sha }}-${{ matrix.node_version }}
+          name: npm-${{ github.event.pull_request.head.sha }}-${{ strategy.job-index }}
           path: ${{ runner.temp }}/npm-pack/*.tgz
           retention-days: 90
 
       - name: Generate docs (if present)
-        if: needs.is_npm.outputs.npm_docs == 'true'
+        if: needs.is_npm.outputs.npm_docs == 'true' && steps.check_engine.outcome == 'success'
         shell: bash
         run: npm run doc
 
       - name: Compress docs
-        if: needs.is_npm.outputs.npm_docs == 'true' && steps.node_versions.outputs.max == matrix.node_version
+        if: needs.is_npm.outputs.npm_docs == 'true' && steps.check_engine.outcome == 'success'
         run: tar --auto-compress -cvf ${{ runner.temp }}/docs.tar.zst ./docs
 
       - name: Upload artifact
-        if: needs.is_npm.outputs.npm_docs == 'true' && steps.node_versions.outputs.max == matrix.node_version
+        if: needs.is_npm.outputs.npm_docs == 'true' && steps.check_engine.outcome == 'success'
         uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # v4.3.4
         with:
-          name: docs-${{ github.event.pull_request.head.sha }}-${{ matrix.node_version }}
+          name: docs-${{ github.event.pull_request.head.sha }}-${{ strategy.job-index }}
```

Comment:
The job-index is relative to the matrix right? Are the jobs always ran in the same order? What could cause it to change? I'm thinking of possible ways an unmerged PR could fail on merge because of a flowzone change between the two events

Reply:
I confirmed the order is always the same and based on the order in which the values are provided in the matrix arrays.